apps/prairielearn/assets/scripts/workspaceClient.ts

```diff
@@ -34,26 +34,38 @@ $(function () {
   });
   const loadingFrame = document.getElementById('loading') as HTMLDivElement;
   const stoppedFrame = document.getElementById('stopped') as HTMLDivElement;
+  const failedFrame = document.getElementById('failed') as HTMLDivElement;
   const workspaceFrame = document.getElementById('workspace') as HTMLIFrameElement;
   const stateBadge = document.getElementById('state') as HTMLSpanElement;
   const messageBadge = document.getElementById('message') as HTMLSpanElement;
+  const failedMessageParagraph = document.getElementById('failed-message') as HTMLElement;
```

Comment:
Nit: drop "paragraph" from the name here? I don't think it's necessary.

Reply:
Done in 688d81d0823225dc61a02bd8fbdf2a4a9f6ce6cb.